src/queues.c

```diff
@@ -198,15 +197,16 @@ int queues_notification_close_id(int id, int reason)
              iter = iter->next) {
                 notification *n = iter->data;
                 if (n->id == id) {
+                        assert(target == NULL);
                         g_queue_remove(waiting, n);
-                        queues_history_push(n);
                         target = n;
                         break;
                 }
         }
 
         if (reason > 0 && reason < 4 && target != NULL) {
```

Comment:
I missed this earlier, if somehow `reason` is outside of the defined values the notification is also never freed.

Reply:
🤦‍♂️ Thank you.